src/service/kafkaRelayPlugin.ts

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+import { Kafka, logLevel, type Producer } from 'kafkajs';
+import { additionalEnvironmentVariables, type Configuration } from '../config';
+import { type ITransportPlugin } from '../interfaces/ITransportPlugin';
+import type { LoggerService } from '@tazama-lf/frms-coe-lib';
+import type { Apm } from '@tazama-lf/frms-coe-lib/lib/services/apm';
+import { validateProcessorConfig } from '@tazama-lf/frms-coe-lib/lib/config/processor.config';
+
+export default class KafkaRelayPlugin implements ITransportPlugin {
+  private readonly kafka: Kafka;
+  private producer?: Producer;
+  private readonly loggerService: LoggerService;
+  private readonly apm: Apm;
+  private readonly configuration: Configuration;
+
+  constructor(loggerService: LoggerService, apm: Apm) {
+    this.loggerService = loggerService;
+    this.apm = apm;
+    // Validate and load configuration using the provided utility
+    this.configuration = validateProcessorConfig(additionalEnvironmentVariables) as Configuration;
+
+    // Use only CA_CERT for TLS
+    const isDev = (this.configuration.nodeEnv ?? 'dev') === 'dev';
+
+    const ssl = isDev
+      ? false
+      : {
+          rejectUnauthorized: false,
+          ca: this.configuration.KAFKA_TLS_CA ? [this.configuration.KAFKA_TLS_CA] : [],
```

Comment:
Value could be set but still be invalid
I suggest looking if something like this is possible
```suggestion
          ca: fs.existsSync(this.configuration.KAFKA_TLS_CA) ? [fs.readFileSync(this.configuration.KAFKA_TLS_CA)] : []
```